ClinGen/clincoded
# Connects to #1169. LOD Calculation for Family Curation

Note: several outstanding bugs are outlined and will be addressed with #1177

![image](https://cloud.githubusercontent.com/assets/4326866/21206570/d6e92242-c217-11e6-8f8d-8f83f14200f9.png)

Implements formulas for calculating LOD scores for Autosomal Dominant and Autosomal Recessive GDMs in the Family Curation page. The field only locks and calculations done automatically for Families in GDMs that fit the above two MOI. Other fields retain the ability for the user to edit the field. 

When the user does not have an entered LOD score (Estimated or Published), regardless of MOI (automatically generated LOD score or not), the dropdown to include the LOD score in final calculations become disabled:
![image](https://cloud.githubusercontent.com/assets/4326866/21206604/14ab24b8-c218-11e6-967e-fd9a07465254.png)

If the user input/generated a LOD score, then selected an option for the LOD score inclusion dropdown, then cleared the LOD score, the LOD score inclusion dropdown also resets.

Testing:

1. Get to the GCI's Family Curation page for a GDM w/ Autosomal Dominant MOI
2. Enter values into the 'For Dominant AND Recessive:
Number of AFFECTED individuals WITH genotype?', 'For Recessive Only:
Number of UNAFFECTED individuals WITHOUT the biallelic genotype?', and 'Number of segregations reported for this Family' fields
3. Set 'Published LOD score' to 'No' and verify that the value in the field and the dropdown below it behave as expected.
4. Change values for fields in step 2 and verify that fields in step 3 work as expected.
5. Repeat steps 2~4 for GDM w/ Autosomal Recessive MOI
6. Repeat steps 2~4 for GDM w/ MOI that is neither of the above


Example scores:
![screen shot 2016-11-11 at 2 45 21 pm](https://cloud.githubusercontent.com/assets/11320314/20233421/27405782-a824-11e6-93e7-33f3d772ed8d.png)
